scripts/check-grid-state-boundaries.sh

```
@@ -0,0 +1,98 @@
+#      state of the moment mark the line with a trailing '// event-time read'.
+#   3. Backends and the adapter stay event-silent: no .subscribe(,
+#      addEventListener( or _broadcastEvent( inside the ViewportService
+#      backends/ and adapter/ directories.
+#   4. Overlays use the per-viewport runtime channel for grid/viewport state;
+#      element.addEventListener( is allowed only in the enumerated files that
+#      listen to cornerstone render events dispatched on the element itself.
+#
+# Run from the repo root: ./scripts/check-grid-state-boundaries.sh
+set -u
+
+cd "$(dirname "$0")/.."
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)